loadtest.py

```diff
@@ -110,4 +110,5 @@ def test_crash_greater_than_20mb_uncompressed():
     resp = utils.post_crash(URL_SERVER, crash_payload)
     print("test_crash_greater_than_20mb_uncompressed: HTTP %s" % resp.status_code)
     # Verify HTTP 413
-    assert 413 == resp.status_code
+
+    assert 413 == resp.status_code, "Failure - HTTP %s, Page text: %s" % (resp.status_code, resp.text)
```

Comment:
OMG--this turned into the most interesting thing ever.

First, I did some debugging and the size of what you're posting is 476 bytes. The dumps are there, but they're empty strings:

```
[th]send: b'POST /submit HTTP/1.1\r\nHost: antenna.dev.mozaws.net\r\nUser-Agent: python-requests/2.12.5\r\nAccept: */*\r\nConnection: keep-alive\r\nAccept-Encoding: gzip, deflate\r\nContent-Type: multipart/form-data; boundary=f3408429c7554ae1991e777e5ea3571c\r\nContent-Length: 476\r\n\r\n'
send: b'--f3408429c7554ae1991e777e5ea3571c\r\nContent-Disposition: form-data; name="ProductName"\r\nContent-Type: text/plain; charset=utf-8\r\n\r\nFirefox\r\n--f3408429c7554ae1991e777e5ea3571c\r\nContent-Disposition: form-data; name="Version"\r\nContent-Type: text/plain; charset=utf-8\r\n\r\n1\r\n--f3408429c7554ae1991e777e5ea3571c\r\nContent-Disposition: form-data; name="upload_file_minidump"; filename="fakecrash.dump"\r\nContent-Type: application/octet-stream\r\n\r\n\r\n--f3408429c7554ae1991e777e5ea3571c--\r\n'
reply: 'HTTP/1.1 200 OK\r\n'
header: Content-Type header: Date header: Content-Length header: Connection test_crash_greater_than_20mb_uncompressed: HTTP 200
```

So then I had a funny feeling and did some poking around. The problem turned out to be really interesting.

First, `utils.assemble_crash_payload` returns a dict. For the dumps, the key/val is the name of the dump `->` `io.BytesIO` of the data. `io.BytesIO` is file-pointer-like in the sense that you `.read()` from it and it returns the data and then the pointer is at the end of the data and future calls to `.read()` return empty strings because there's nothing left to consume.

So `utils.multipart_encode` goes through the `crash_payload` dict and generates the POST data. As it iterates through the dumps, it calls `.read()` thus consuming all that data. The code checks the result, but otherwise throws it out.

So `utils.post_data` is called and that internally calls `utils.multipart_encode` which calls `.read()` on all those dumps again, but this time they return empty strings and you end up with a payload with empty-string dumps in it and that's what gets posted to Antenna and since that's fine, it gets back an HTTP 200.

This is actually a problem with all the tests in the file that have that additional `if len(payload) != size` section.